### ISSUE TITLE
.NET MAUI Blazor - additional template options

### ISSUE THREAD

**Issue by Dukenware** [dotnet/maui#4488]
### Description

Would it be possible to add a .NET MAUI Blazor template with options "ASP.NET Core hosted" and "Authentication" that we get when creating a new Blazor WebAssemlby project?

### Public API Changes

Not applicable

### Intended Use-Case

My idea behind this is to have a project template for an application that can be run as WebApplication just the same as a .NET MAUI Blazor application with working authentication/authorization and an API to work against.

**Comment by Eilon** [comment]
Right, if there was an auth option, it couldn't just re-use what the ASP.NET Blazor templates have. It would instead use MAUI Essentials (or similar) to do a "native" auth flow (because this _is_ a native app!).

**Comment by Ghevi** [comment]
@Eilon I'm not sure if this is relevant, but I was able to add `CascadingAuthenticationState` in the `Main.razor` file and then using `[CascadingParameter] protected Task<AuthenticationState> AuthenticationStateTask { get; set; }` where needed.
The `CustomAuthProvider`  has methods that call the server to do the authentication with Jwt.
It inherits from `AuthenticationStateProvider` and it's registered it in `MauiProgram.cs` with
`services.AddOptions();`
 `services.AddAuthorizationCore(options =>
{
    // add your policies
});`
`services.AddScoped<CustomAuthProvider>();`
`services.AddScoped<AuthenticationStateProvider>(provider => provider.GetRequiredService<CustomAuthProvider>());`

 Is this unrecommended?

**Comment by Eilon** [comment]
@Ghevi it could work just fine, depending on what the custom auth provider does. I think in the case of ASP.NET there are request/responses happening that I thing would not work the same way (or at all) in a .NET MAUI app.

**Comment by mattleibow** [comment]
This is now done. We have some more blazor templates - specifically the mobile+web hybrid.

We are looking into auth, and that is tracked in an internal project: https://github.com/orgs/dotnet/projects/290?pane=issue&itemId=94966469

For any other specific options, please open a specific option so we can evaluate and work.
